main.cpp

```diff
@@ -40,9 +40,151 @@ send me a DM to check your pull request
 
  Wait for my code review.
  */
+#include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs);
+    float substract(float lhs, float rhs);
+    float multiply(float lhs, float rhs );
+    float divide(float lhs, float rhs );
+};
+
+float FloatType::add(float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::substract(float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs)
+{
+    if (rhs == 0.0f)
+    {
+        std::cout << "Warning! You are trying to divide a float type by zero!" << std::endl;
+    }
+    
+    return lhs / rhs;
+}
+
+
+struct DoubleType
+{
+    double add(double lhs, double rhs);
+    double substract(double lhs, double rhs);
+    double multiply(double lhs, double rhs );
+    double divide(double lhs, double rhs );
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs;
+}
+
+double DoubleType::substract(double lhs, double rhs)
+{
+    return lhs - rhs;
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs)
+{
+    return lhs / rhs;
+}
```

Comment:
- [ ] you forgot the warning that `FloatType::divide` has

Reply:
I check it and it's there, the lines you are pointing to are from the Double Type